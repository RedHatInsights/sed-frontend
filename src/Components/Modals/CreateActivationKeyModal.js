import * as React from 'react';
import { Modal, ModalVariant } from '@patternfly/react-core';
import CreateActivationKeyForm from '../Forms/CreateActivationKeyForm';
import useCreateActivationKey from '../../hooks/useCreateActivationKey';
import propTypes from 'prop-types';
import Loading from '../LoadingState/Loading';
import { useQueryClient } from 'react-query';

const CreateActivationKeyModal = (props) => {
  const queryClient = useQueryClient();
  const [created, setCreated] = React.useState(false);
  const [error, setError] = React.useState(false);
  const { handleModalToggle, isOpen } = props;
  const { mutate, isLoading } = useCreateActivationKey();
  const submitForm = (name, role, serviceLevel, usage) => {
    mutate(
      { name, role, serviceLevel, usage },
      {
        onSuccess: () => {
          setError(false);
          setCreated(true);
          queryClient.invalidateQueries('activation_keys');
        },
        onError: () => {
          setError(true);
          setCreated(false);
        },
      }
    );
  };
  return (
    <Modal
      variant={ModalVariant.large}
      title="Create new activation key"
      description=""
      isOpen={isOpen}
      onClose={handleModalToggle}
    >
      {isLoading ? (
        <Loading />
      ) : (
        <CreateActivationKeyForm
          handleModalToggle={handleModalToggle}
          submitForm={submitForm}
          isSuccess={created}
          isError={error}
        />
      )}
    </Modal>
  );
};

CreateActivationKeyModal.propTypes = {
  handleModalToggle: propTypes.func.isRequired,
  isOpen: propTypes.bool.isRequired,
};

export default CreateActivationKeyModal;
