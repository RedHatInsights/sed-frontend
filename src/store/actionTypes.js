// Current state actions
export const GET_CURR_STATE = 'GET_CURR_STATE';
export const SET_CURR_STATE = 'SET_CURR_STATE';

// Connected hosts
export const GET_CONNECTED_HOSTS = 'GET_CONNECTED_HOSTS';
