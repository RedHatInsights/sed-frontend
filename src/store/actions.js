import { useConfigApi, useInventoryApi } from '../api';
import {
  GET_CURR_STATE,
  GET_LOG,
  SET_CURR_STATE,
  GET_CONNECTED_HOSTS,
} from './actionTypes';

const fetchCurrState = (api) => () => ({
  type: GET_CURR_STATE,
  payload: api.getProfile('current'),
});

const saveCurrState =
  (api) =>
  ({ compliance, remediations, active }) => ({
    type: SET_CURR_STATE,
    payload: api.createProfile({
      compliance,
      insights: true,
      remediations,
      active,
    }),
  });

const fetchLog =
  (api) =>
  ({ sort = 'created_at:desc', perPage = 50, page = 1 } = {}) => ({
    type: GET_LOG,
    payload: api.getProfiles(sort, perPage, (page - 1) * perPage),
  });

const fetchConnectedHosts = (api) => () => ({
  type: GET_CONNECTED_HOSTS,
  payload: api.getConnectedHosts(),
});

export const useActions = () => {
  const api = useConfigApi();
  const inventoryApi = useInventoryApi();

  return {
    fetchConnectedHosts: fetchConnectedHosts(inventoryApi),
    fetchLog: fetchLog(api),
    saveCurrState: saveCurrState(api),
    fetchCurrState: fetchCurrState(api),
  };
};
