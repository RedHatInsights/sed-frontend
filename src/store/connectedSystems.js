import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { GET_CONNECTED_HOSTS } from './actionTypes';

const initialState = {
  loaded: false,
  values: {},
};

const connectedSystemsPending = (state) => ({
  ...state,
  loaded: false,
});

const connectedSystemsFulfilled = (state, { payload }) => ({
  ...state,
  loaded: true,
  hosts: payload?.results || [],
  total: payload?.count || 0,
  page: payload?.page || 0,
  perPage: payload?.per_page || 0,
});

export default applyReducerHash(
  {
    [`${GET_CONNECTED_HOSTS}_PENDING`]: connectedSystemsPending,
    [`${GET_CONNECTED_HOSTS}_FULFILLED`]: connectedSystemsFulfilled,
    [`${GET_CONNECTED_HOSTS}_ERROR`]: () => ({ error: true }),
  },
  initialState
);
