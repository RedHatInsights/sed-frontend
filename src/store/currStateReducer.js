import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { GET_CURR_STATE, SET_CURR_STATE } from './actionTypes';

const initialState = {
  loaded: false,
  values: {},
};

const currStatePending = (state) => ({
  ...state,
  loaded: false,
});
const currStateFulfilled = (state, { payload }) => ({
  ...state,
  loaded: true,
  values: {
    useOpenSCAP: payload?.state?.compliance_openscap === 'enabled',
    enableCloudConnector: payload?.state?.remediations === 'enabled',
    hasInsights: payload?.state?.insights === 'enabled',
  },
});

export default applyReducerHash(
  {
    [`${SET_CURR_STATE}_PENDING`]: currStatePending,
    [`${GET_CURR_STATE}_PENDING`]: currStatePending,
    [`${GET_CURR_STATE}_FULFILLED`]: currStateFulfilled,
    [`${SET_CURR_STATE}_FULFILLED`]: currStateFulfilled,
    [`${GET_CURR_STATE}_ERROR`]: () => ({ error: true }),
  },
  initialState
);
