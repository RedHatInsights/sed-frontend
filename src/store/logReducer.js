import { applyReducerHash } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import { GET_LOG } from './actionTypes';

const initialState = {
  loaded: false,
  results: [],
};

const logPending = (state) => ({
  ...state,
  loaded: false,
});
const logFulfilled = (state, { payload }) => ({
  ...state,
  loaded: true,
  ...(payload || {}),
});

export default applyReducerHash(
  {
    [`${GET_LOG}_PENDING`]: logPending,
    [`${GET_LOG}_FULFILLED`]: logFulfilled,
    [`${GET_LOG}_ERROR`]: () => ({ error: true }),
  },
  initialState
);
