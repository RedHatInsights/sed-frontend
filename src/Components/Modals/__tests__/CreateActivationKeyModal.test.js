import React from 'react';
import { Provider } from 'react-redux';
import { init } from '../../../store';
import { QueryClient, QueryClientProvider } from 'react-query';
import { render } from '@testing-library/react';
import CreateActivationKeyModal from '../CreateActivationKeyModal';

const queryClient = new QueryClient();
const registry = init();

describe('Create Activation Key Modal', () => {
  it('renders correctly', () => {
    const props = { handleModalToggle: jest.fn(), isOpen: true };
    render(
      <Provider store={registry.getStore()}>
        <QueryClientProvider client={queryClient}>
          <CreateActivationKeyModal {...props} />
        </QueryClientProvider>
      </Provider>
    );
    /**
     * document.body needed because the modal
     * does not render within the container
     *
     */
    expect(document.body).toMatchSnapshot();
  });
});
