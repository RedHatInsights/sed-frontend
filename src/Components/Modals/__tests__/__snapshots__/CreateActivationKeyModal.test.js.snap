// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Activation Key Modal renders correctly 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  />
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-describedby="pf-modal-part-2"
          aria-labelledby="pf-modal-part-1"
          aria-modal="true"
          class="pf-c-modal-box pf-m-lg"
          data-ouia-component-id="OUIA-Generated-Modal-large-1"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-0"
          role="dialog"
        >
          <button
            aria-disabled="false"
            aria-label="Close"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Modal-large-1-ModalBoxCloseButton"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 352 512"
              width="1em"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              />
            </svg>
          </button>
          <header
            class="pf-c-modal-box__header"
          >
            <h1
              class="pf-c-modal-box__title"
              id="pf-modal-part-1"
            >
              
              <span
                class="pf-c-modal-box__title-text"
              >
                Create new activation key
              </span>
            </h1>
            
          </header>
          <div
            class="pf-c-modal-box__body"
            id="pf-modal-part-2"
          >
            <form
              class="pf-c-form"
              id="activation-key-form"
              novalidate=""
            >
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      Name
                    </span>
                    <span
                      aria-hidden="true"
                      class="pf-c-form__label-required"
                    >
                       
                      *
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <input
                    aria-invalid="false"
                    class="pf-c-form-control"
                    data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                    data-ouia-component-type="PF4/TextInput"
                    data-ouia-safe="true"
                    id="activation-key-name"
                    label="Name"
                    name="name"
                    required=""
                    type="text"
                    value=""
                  />
                  <div
                    aria-live="polite"
                    class="pf-c-form__helper-text"
                    id="undefined-helper"
                  >
                    Provide a name to be used when registering the activation key. Your activation key name must be unique, may contain only numbers, letters, underscores, and hyphens, and less than 256 characters.
                  </div>
                </div>
              </div>
              <div
                class="pf-c-form__group pf-m-action"
              >
                <div
                  class="pf-c-form__group-control"
                >
                  <div
                    class="pf-c-form__actions"
                  >
                    <button
                      aria-disabled="true"
                      class="pf-c-button pf-m-primary pf-m-disabled"
                      data-ouia-component-id="OUIA-Generated-Button-primary-1"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      data-testid="activation-key-submit-button"
                      disabled=""
                      id="submit-activation-key-button"
                      type="submit"
                    >
                      Create
                    </button>
                    <button
                      aria-disabled="false"
                      class="pf-c-button pf-m-link"
                      data-ouia-component-id="OUIA-Generated-Button-link-1"
                      data-ouia-component-type="PF4/Button"
                      data-ouia-safe="true"
                      id="cancel-activation-key-button"
                      type="button"
                    >
                      Cancel
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Create Activation Key Modal renders the wizard when feature flag is enabled 1`] = `
<body
  class="pf-c-backdrop__open"
>
  <div
    aria-hidden="true"
  />
  <div>
    <div
      class="pf-c-backdrop"
    >
      <div
        class="pf-l-bullseye"
      >
        <div
          aria-label="Create activation key wizard"
          aria-labelledby="pf-modal-part-1"
          aria-modal="true"
          class="pf-c-modal-box pf-m-lg"
          data-ouia-component-id="OUIA-Generated-Modal-large-2"
          data-ouia-component-type="PF4/ModalContent"
          data-ouia-safe="true"
          id="pf-modal-part-1"
          role="dialog"
        >
          
          <div
            class="pf-c-wizard"
            style="height: 400px;"
          >
            <div
              class="pf-c-wizard__header"
            >
              <button
                aria-disabled="false"
                aria-label="Close"
                class="pf-c-button pf-m-plain pf-c-wizard__close"
                data-ouia-component-id="OUIA-Generated-Button-plain-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  />
                </svg>
              </button>
              <h2
                aria-label="Create activation key"
                class="pf-c-title pf-m-3xl pf-c-wizard__title"
                data-ouia-component-id="OUIA-Generated-Title-1"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
                id="pf-wizard-title-0"
              >
                Create activation key
              </h2>
              
            </div>
            <button
              aria-expanded="false"
              aria-label="Wizard Toggle"
              class="pf-c-wizard__toggle"
            >
              <span
                class="pf-c-wizard__toggle-list"
              >
                <span
                  class="pf-c-wizard__toggle-list-item"
                >
                  <span
                    class="pf-c-wizard__toggle-num"
                  >
                    1
                  </span>
                   
                  Name
                </span>
              </span>
              <span
                class="pf-c-wizard__toggle-icon"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 320 512"
                  width="1em"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="pf-c-wizard__outer-wrap"
            >
              <div
                class="pf-c-wizard__inner-wrap"
              >
                <nav
                  aria-label="Create activation key steps"
                  aria-labelledby="pf-wizard-title-0"
                  class="pf-c-wizard__nav"
                  data-ouia-component-id="OUIA-Generated-WizardNav-1"
                  data-ouia-component-type="PF4/WizardNav"
                  data-ouia-safe="true"
                >
                  <ol
                    class="pf-c-wizard__nav-list"
                  >
                    <li
                      class="pf-c-wizard__nav-item"
                    >
                      <button
                        aria-current="step"
                        class="pf-c-wizard__nav-link pf-m-current"
                        data-ouia-component-id="OUIA-Generated-WizardNavItem-1"
                        data-ouia-component-type="PF4/WizardNavItem"
                        data-ouia-safe="true"
                      >
                        Name
                      </button>
                    </li>
                    <li
                      class="pf-c-wizard__nav-item"
                    >
                      <button
                        aria-current="false"
                        aria-disabled="true"
                        class="pf-c-wizard__nav-link pf-m-disabled"
                        data-ouia-component-id="OUIA-Generated-WizardNavItem-2"
                        data-ouia-component-type="PF4/WizardNavItem"
                        data-ouia-safe="true"
                        disabled=""
                        id="1"
                      >
                        Workload
                      </button>
                    </li>
                    <li
                      class="pf-c-wizard__nav-item"
                    >
                      <button
                        aria-current="false"
                        aria-disabled="true"
                        class="pf-c-wizard__nav-link pf-m-disabled"
                        data-ouia-component-id="OUIA-Generated-WizardNavItem-3"
                        data-ouia-component-type="PF4/WizardNavItem"
                        data-ouia-safe="true"
                        disabled=""
                        id="2"
                      >
                        System purpose
                      </button>
                    </li>
                    <li
                      class="pf-c-wizard__nav-item"
                    >
                      <button
                        aria-current="false"
                        aria-disabled="true"
                        class="pf-c-wizard__nav-link pf-m-disabled"
                        data-ouia-component-id="OUIA-Generated-WizardNavItem-4"
                        data-ouia-component-type="PF4/WizardNavItem"
                        data-ouia-safe="true"
                        disabled=""
                        id="3"
                      >
                        Review
                      </button>
                    </li>
                  </ol>
                </nav>
                <div
                  aria-label="Create activation key content"
                  aria-labelledby="pf-wizard-title-0"
                  class="pf-c-wizard__main"
                >
                  <div
                    class="pf-c-wizard__main-body"
                  >
                    <h2
                      class="pf-c-title pf-m-xl pf-u-mb-sm"
                      data-ouia-component-id="OUIA-Generated-Title-2"
                      data-ouia-component-type="PF4/Title"
                      data-ouia-safe="true"
                    >
                      Name key
                    </h2>
                    <p
                      class="pf-u-mb-xl"
                      data-ouia-component-id="OUIA-Generated-Text-1"
                      data-ouia-component-type="PF4/Text"
                      data-ouia-safe="true"
                      data-pf-content="true"
                    >
                      This name cannot be modified after the activation key is created.
                    </p>
                    <form
                      class="pf-c-form"
                      novalidate=""
                    >
                      <div
                        class="pf-c-form__group"
                      >
                        <div
                          class="pf-c-form__group-label"
                        >
                          <label
                            class="pf-c-form__label"
                            for="activation-key-name"
                          >
                            <span
                              class="pf-c-form__label-text"
                            >
                              Name
                            </span>
                            <span
                              aria-hidden="true"
                              class="pf-c-form__label-required"
                            >
                               
                              *
                            </span>
                          </label>
                           
                        </div>
                        <div
                          class="pf-c-form__group-control"
                        >
                          <input
                            aria-invalid="false"
                            class="pf-c-form-control"
                            data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                            data-ouia-component-type="PF4/TextInput"
                            data-ouia-safe="true"
                            id="activation-key-name"
                            required=""
                            type="text"
                            value=""
                          />
                          <div
                            aria-live="polite"
                            class="pf-c-form__helper-text"
                            id="activation-key-name-helper"
                          >
                            Your activation key name must be unique and must contain only numbers, letters, underscores, and hyphens.
                          </div>
                        </div>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
              <footer
                class="pf-c-wizard__footer"
              >
                <button
                  aria-disabled="true"
                  class="pf-c-button pf-m-primary pf-m-disabled"
                  data-ouia-component-id="OUIA-Generated-Button-primary-2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="submit"
                >
                  Next
                </button>
                <button
                  aria-disabled="true"
                  class="pf-c-button pf-m-secondary pf-m-disabled"
                  data-ouia-component-id="OUIA-Generated-Button-secondary-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  Back
                </button>
                <div
                  class="pf-c-wizard__footer-cancel"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-link"
                    data-ouia-component-id="OUIA-Generated-Button-link-2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    Cancel
                  </button>
                </div>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
